node/test/util/push.js

```diff
@@ -33,8 +33,160 @@
 const co     = require("co");
 
 const Push            = require("../../lib/util/push");
+const RepoAST         = require("../../lib/util/repo_ast");
+const RepoASTUtil     = require("../../lib/util/repo_ast_util");
 const RepoASTTestUtil = require("../../lib/util/repo_ast_test_util");
 
+/**
+ * Return a map from name to RepoAST that is the same as the one in the
+ * specified `expected`, but having transformed the name of each reference from
+ * the form of `ref/commit/<logical commit id>` to
+ * `ref/commit/<physical commit id>`, where <logical commit id> is the id of
+ * the commit the ref points to.  The specified `mapping` object contains a
```

Comment:
I feel that "object" is more natural as I am discussing the value that was passed in an actual call rather than an idealized argument.